### PR TITLE
Support OpenId credential for google identification

### PR DESCRIPTION
**Pull request by clementetb**
Allow to login with google's `id_token` param. This is an additive login method that should work alongside with `AuthCode`.

This PR is still pending on Core to release the feature.

### CODE REVIEW

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/auth/GoogleAuthType.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package io.realm.mongodb.auth;
+
+public enum GoogleAuthType {
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/auth/GoogleAuthType.java:3]: We need public API docs for this, including each enum type + missing license headers

File: realm/realm-library/src/objectServer/java/io/realm/mongodb/Credentials.java [hunk 2]
```diff
@@ -150,17 +152,33 @@ public static Credentials facebook(String accessToken) {
     }
 
     /**
-     * Creates credentials representing a login using a Google Authorization Code.
+     * Creates credentials representing a login using a Google access token of a given {@link GoogleAuthType}.
+     * <p>
+     * This provider must be enabled on MongoDB Realm to work.
+     *
+     * @param accessToken the access token returned when logging in to Google.
```
Comment (cmelchior) [inline realm/realm-library/src/objectServer/java/io/realm/mongodb/Credentials.java:159]: We should probably just rename this to `token` as the type is determined by the enum?

**Review by cmelchior — CHANGES_REQUESTED** [review]
Thanks for adding this 💯 . I added a few comments. Also, I don't think this should target releases as it isn't a bug fix as such? Also, we need a changelog update

File: realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/internal/network/LoggingInterceptorTest.kt [hunk 3]
```diff
@@ -204,22 +205,55 @@ class LoggingInterceptorTest {
...
     }
 
+
+    @Test
+    fun googleTokenLogin_idToken_noObfuscation() {
+        app = TestApp()
+        testLogger = getLogger()
+        val token = "google-token"
+
+        try {
```
Comment (rorbech) [inline realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/internal/network/LoggingInterceptorTest.kt:231]: There is an `assertFailsWithErrorCode` for this pattern. There is still no functionality to match the error though, so you will still have to inspect the error message with `.also { e -> e.message.contains}` or similar.

**Comment by cmelchior** [comment]
Ideally this PR shoul have been targeted `master`, but since we are going to make a release immediately it is probably not worth fixing pointing it towards the correct branch. So just keeping the base as `releases` for now.

File: CHANGELOG.md [hunk 4]
```diff
@@ -46,6 +56,11 @@
 * APIs are backwards compatible with all previous release of realm-java in the 10.x.y series.
 * Realm Studio 10.0.0 or above is required to open Realms created by this version.
 
+### Internal
+* Updated to Realm Sync: 10.1.3.
+* Updated to Realm Core: 10.1.3.
+* Updated to Object Store commit: fc6daca61133aa9601e4cb34fbeb9ec7569e162e.
+
 
 ## 10.1.0 (2020-10-23)
```
[the remark this answers is not in the document]
Comment (rorbech) [reply CHANGELOG.md:65]: Seems like master hasn't been merged. Shouldn't we do that on a minor release? Don't know how many changes there are though.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,19 +1,29 @@
...
 
 ### Breaking Changes
 * None.
 
 ### Enhancements
-* None.
+* [RealmApp] Added `Credentials.google(token: String, authType: GoogleAuthType)`, as MongoDB Realm support multiple ways of logging in Google Accounts.
 
 ### Fixes
+* [RealmApp] Bug that would prevent eventual consistency during conflict resolution. Affected clients would experience data divergence and potentially consistency errors as a result if they experienced conflict resolution between cycles of Create-Erase-Create for objects with primary keys.
```
Comment (rorbech) [inline CHANGELOG.md:13]: A bit convoluted. Tried to rephrase but failed as I was a bit worried that I changed the meaning. Maybe just re-read to verify.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,19 +1,29 @@
...
+* Complementary fix for missed edge case in https://github.com/realm/realm-java/pull/7220 where KAPT crash if we process a RealmObject referencing a type in RealmList defined in another module. (Issue [#7213](https://github.com/realm/realm-java/issues/7213), since v10.0.0).
 * Clean up JNI references to prevent crash from JNI reference table overflow (Issue [#7217](https://github.com/realm/realm-java/issues/7217))
 
 ### Compatibility
 * File format: Generates Realms with format v20. Unsynced Realms will be upgraded from Realm Java 2.0 and later. Synced Realms can only be read and upgraded if created with Realm Java v10.0.0-BETA.1.
 * APIs are backwards compatible with all previous release of realm-java in the 10.x.y series.
 * Realm Studio 10.0.0 or above is required to open Realms created by this version.
 
+### Internal
+* Updated to Realm Sync: 10.1.4.
```
Comment (rorbech) [inline CHANGELOG.md:23]: I guess we should also included the Sync _protocol 2_ snippet

**Review by rorbech — CHANGES_REQUESTED** [review]
Seems good - Only some minor comments, but there are still missing docs and a "figure out why this is failing"

File: CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (rorbech) [reply CHANGELOG.md:65]: I guess this wasn't the release PR, sorry.

File: CHANGELOG.md (hunk 5 above)
Comment (cmelchior) [reply CHANGELOG.md:13]: This is a copy from the Sync changelog. I agree it is a bit convoluted, but so was the error, so not sure there is much we can do to rephrase it 🤔

Comment (rorbech) [reply CHANGELOG.md:13]: Heh. I didn't even get that this was actually the _protocol 2_ snippet.

**Review by rorbech — APPROVED** [review]

**Pull request merged**